### PR TITLE
batcheval: allow writing tombstones in `AddSSTable`

### LINKED ISSUE #82922: batcheval: `AddSSTable` must allow tombstones
For cluster-to-cluster replication, `AddSSTable` must allow writing tombstones.

Jira issue: CRDB-16732

Epic CRDB-2624

### PR DESCRIPTION
**Pull request by erikgrinaker**
This patch allows writing MVCC point tombstones in `AddSSTable`, which
will be needed by cluster-to-cluster replication. Previously, SSTs
containing tombstones could error or be silently accepted. The version
gate `AddSSTableTombstones` has been added to allow clients to rely on
this behavior.

Note that `AddSSTable` has never made any attempt at correctly
calculating `MVCCStats.GCBytesAge` when shadowing existing data, and
this patch does not attempt to either. This will be addressed later.

Resolves #82922.
Touches #82920.

Release note: None

### CODE REVIEW

File: pkg/kv/kvserver/batcheval/cmd_add_sstable.go [hunk 1]
```diff
@@ -429,31 +429,21 @@ func assertSSTContents(sst []byte, sstTimestamp hlc.Timestamp, stats *enginepb.M
 
 	// Check SST KV pairs.
 	iter.SeekGE(storage.MVCCKey{Key: keys.MinKey})
```
Comment (aliher1911) [inline pkg/kv/kvserver/batcheval/cmd_add_sstable.go:431]: Looks like leftover since we have it as part of for loop now.

**Review by aliher1911 — APPROVED** [review]
LGTM once the comment is resolved.

File: pkg/kv/kvserver/batcheval/cmd_add_sstable.go (hunk 1 above)
Comment (erikgrinaker) [reply pkg/kv/kvserver/batcheval/cmd_add_sstable.go]: Oops, nice catch -- thank you!

**Comment by erikgrinaker** [comment]
> We may want to consider a version gate here, since 22.1 will sometimes error when ingesting tombstones

Meh, added a version gate `AddSSTableTombstones` for this, just in case.

@gh-casper Just a heads-up that 22.1 nodes don't really support writing MVCC _point_ tombstones in `AddSSTable`. I doubt that we'll ever end up trying to ingest stuff from cluster replication into a 22.1 node, but out of an abundance of caution, there's a version gate `AddSSTableTombstones` that can be checked to make sure the server will support ingesting tombstones.

**Comment by erikgrinaker** [comment]
CI failure appears unrelated.

bors r=aliher1911

**Pull request merged**